### DISCUSSION TITLE
ROG Ally X

### DISCUSSION THREAD

**Discussion by notSean1992** [seerge/g-helper#2925] [Q&A]
Is there any plans to support the ROG Ally X? Installed it just now and unfortunately embedded controller is not recognised. Seems Handheld Companion has a similar problem.

**Comment by seerge** [comment]
@Seanster911 Ally X should be supported, what exactly is "not recognised" ? 

If something is not working as you expect it to, please [post an issue](https://github.com/seerge/g-helper/issues/new?assignees=&labels=&projects=&template=bug_report.yml) and fill all asked questions there. Thanks.

**Comment by notSean1992** [comment]
The aura lights around the joysticks do not work, the buttons do not register any presses including the triggers and analogue sticks (have checked in calibration on steam and bindings in helper).

I will fill out an issue later when I can get the log file. Thanks!

**Reply by seerge** [reply]
@Seanster911 controller is handled directly by windows itself (it identifies itself as XBOX controller), not by G-Helper.

You can only rebind buttons / triggers in the app (same as in AC), but if controller in general doesn't work - then problem is somewhere else :) Make sure you have all drivers installed and controller is not disabled anyhow ?

**Reply by notSean1992** [reply]
Strange, it stops working as soon as I hit stop ASUS services in ghelper and starts working again once I restart them.

**Reply by seerge** [reply]
@Seanster911 most likely asus services just disable controller before stopping by some reason.

Post a full log from the `%appdata%\GHelper` and also download https://www.nirsoft.net/utils/usbdeview-x64.zip

And then find ASUS devices (Vendor ID `0x0b05`) and post a screenshot here.
![image](https://github.com/user-attachments/assets/668e685f-682f-4157-8840-3c3c1207d48a)

**Reply by notSean1992** [reply]
![image](https://github.com/user-attachments/assets/64e1758f-5370-44b0-af6f-49138d27e417)
[log.txt](https://github.com/user-attachments/files/16494210/log.txt)


Here's the screenshot and the log. Thanks for the help. Want me to re-raise it as an issue for you or is this enough?

**Reply by seerge** [reply]
@Seanster911 ok, thanks. I have added `1b4c` device ID to the list as well. Try this build and post a log from it

[GHelper.zip](https://github.com/user-attachments/files/16494315/GHelper.zip)

If controller is still completely disabled, try to open controller settings from G-Helper and uncheck / check toggle there.

**Reply by notSean1992** [reply]
This seems to have got it working perfectly. Thanks for all the help!

**Reply by seerge** [reply]
@Seanster911 ok! That was easy.

Good to know. Is everything else working as it should then (i assume you had regular Ally as well and used G-Helper before?)

**Reply by notSean1992** [reply]
I didn't have an Ally before, just really hate AC. Everything seems to be working as intended, all the binding and things are exactly as expected. Will post an issue if anything else shows up.

**Comment by lgecse** [comment]
@seerge Great work, I love GHelper. I'd also add that the power profiles don't seem to be synced like on the OG Ally. I see all 3 profiles, but the limits are set to 30W, and the sliders are also different. 

On Ally X I see 2 sliders only: Platform (CPU + GPU) (30W in every profile) and a second one for the CPU only, this defaults everywhere to 80W. I cannot set at the moment sPPT or fPPT values.

**Reply by seerge** [reply]
@lgecse hello, can you upload log from the application ? As for profiles, they are stored in BIOS, not in the app. Power limits slider positions don't have any effect unless you check `Apply` under `Fans + Power`.

**Reply by lgecse** [reply]
@seerge sure: 

<img width="837" alt="image" src="https://github.com/user-attachments/assets/6a2dc2b6-c735-4d33-b0ca-ab6d9a26d131">
<img width="823" alt="image" src="https://github.com/user-attachments/assets/af99edbe-87f5-488e-b02c-7f9d1295f95c">


It can also throw errors when touching the CPU:
<img width="395" alt="image" src="https://github.com/user-attachments/assets/3c69df76-b835-4ca1-b4c7-f2355715528a">

Overall after setting values (with platform slider it's aligning CPU as well) it works. Restarting the app keeps previously set wattage values. However after removing the config file I see again in every profile the 30-80 default values. 
Log with setting limit to 13: 
[log.txt](https://github.com/user-attachments/files/16502690/log.txt)

I might not remember well how this was on OG Ally, but as I recall on the first start I have seen the AC default wattages on the profiles (10-15-30). I expected GHelper to show the X values (13-17-30). Again, I might be wrong here.

**Reply by seerge** [reply]
@lgecse try this build

[GHelper.zip](https://github.com/user-attachments/files/16502900/GHelper.zip)

**Reply by lgecse** [reply]
I see no difference. I found this in the code, for the ALLY X we have this modeB0 set to true:

https://github.com/seerge/g-helper/blob/a83b42c26310e836188c2b11c0e36f3c01432d4e/app/Fans.cs#L919

**Reply by seerge** [reply]
@lgecse it checks (based on available endpoints), if it's All AMD model (i.e. G14 2022, where power sliders mean something else). Ally X just has a strange reporting here so I have manually added exception here 

https://github.com/seerge/g-helper/commit/a83b42c26310e836188c2b11c0e36f3c01432d4e

So it should be False in your case. May be last comment contains a wrong build by some reason, try again this one
[GHelper.zip](https://github.com/user-attachments/files/16503080/GHelper.zip)

**Reply by lgecse** [reply]
Yup, this is it, I can see the same settings as on the OG Ally. Much appreciated!

**Reply by seerge** [reply]
@lgecse ok, great. 

By some reason Ally X seem to report existence of `` PPT_CPUB0 = 0x001200B0;  // CPU PPT on 2022 (PPT_LIMIT_APU)`` which confused G-Helper in a first place, as that endpoint exists only on "Advantage Edition" laptops only (AMD CPU + AMD dGPU)

**Reply by seerge** [reply]
@lgecse hello, someone random has made a comment on youtube that they can't bind a Command Center button to something under `Extra` on Ally X. Can you check if it works or not (i usually bind it to toggle modes for example) ? 

Thanks.

**Reply by lgecse** [reply]
Hey @seerge 

I can share screenshots later, but I have the armoury crate button mapped to show/hide GHeleper, and the command center button to switch between mouse and controller mode. Both works as expected on my end.

**Reply by seerge** [reply]
@lgecse ok, amazing! Then apparently it's just another false claim. I just thought that CommandCenter button keycode could change in Ally X.

**Reply by lgecse** [reply]
hey @seerge, FYI, I've just changed the button behaviour to switch between modes, and it also works as expected.

**Reply by seerge** [reply]
@lgecse ok, thanks for confirming :)
